test/jasmine/tests/calcdata_test.js

```diff
@@ -869,6 +871,216 @@ describe('calculated data and points', function() {
+            var i, j, k;
+            for(i = 0; i < traces.length; i++) {
+                tracesSchema.push(schema.traces[traces[i]]);
+            }
+            var cartesianTraces = tracesSchema.filter(function(t) {
+                return t.categories.length && t.categories.indexOf('cartesian') !== -1;
+            });
+
+            // excludedTraces are traces that do not support sorting by value
+            var excludedTraces = [ 'carpet', 'contourcarpet',
+                // TODO: add support for the following
+                'histogram2dcontour'];
+
+            var supportedCartesianTraces = cartesianTraces.filter(function(t) {
+                if(excludedTraces.indexOf(t.type) === -1) return true;
+            });
+
+            var cat = ['a', 'b', 'c'];
+
+            // oneOrientationTraces are traces for which swapping x/y is not supported
+            var oneOrientationTraces = ['ohlc', 'candlestick'];
+
+            function makeData(type, a, b, axId) {
+                var input = [a, b];
+                var cat = input[axId === 'yaxis' ? 1 : 0];
+                var data = input[axId === 'yaxis' ? 0 : 1];
+
+                var measure = [];
+                for(j = 0; j < data.length; j++) {
+                    measure.push('absolute');
+                }
+
+                var z = Lib.init2dArray(cat.length, data.length);
+                for(j = 0; j < z.length; j++) {
+                    for(k = 0; k < z[j].length; k++) {
+                        z[j][k] = 0;
+                    }
+                }
+                if(axId === 'xaxis') {
+                    for(j = 0; j < b.length; j++) {
+                        z[0][j] = b[j];
+                    }
+                }
+                if(axId === 'yaxis') {
+                    for(j = 0; j < b.length; j++) {
+                        z[j][0] = b[j];
+                    }
+                }
+
+                return Lib.extendDeep({}, {
+                    orientation: axId === 'yaxis' ? 'h' : 'v',
+                    type: type,
+                    x: cat,
+                    a: cat,
+
+                    b: data,
+                    y: data,
+                    z: z,
+
+                    // For OHLC
+                    open: data,
+                    close: data,
+                    high: data,
+                    low: data,
+
+                    // For waterfall
+                    measure: measure,
+
+                    // For splom
+                    dimensions: [
+                        {
+                            label: 'DimensionA',
+                            values: a
+                        },
+                        {
+                            label: 'DimensionB',
+                            values: b
+                        }
+                    ]
+                });
+            }
+
+            supportedCartesianTraces.forEach(function(trace) {
+                ['xaxis', 'yaxis'].forEach(function(axId) {
```

Comment:
`axId` is usually reserved for `x2`, `y4`, etc. For `xaxis` and `yaxis` we usually use `axName`

Reply:
Much clearer thanks! Done in 2d379f7647bec909b806f56276b6f6225578bc6b